azimuth/startup.py

```diff
@@ -146,16 +183,22 @@ def make_startup_tasks(
     dms = {k: v for k, v in dataset_split_managers.items() if v is not None}
     tasks: Dict[DatasetSplitName, DaskModule] = {}
 
-    for dataset_split_name, dm in dms.items():
+    if DatasetSplitName.all not in dataset_split_names:
+        available_datasets_splits = list(set(dataset_split_names) & set(dms))
+    else:
+        available_datasets_splits = [DatasetSplitName.all]
+
+    for dataset_split_name in available_datasets_splits:
```

Comment:
Hfff, this dms logic is becoming messy, don't you agree? Here is an idea
```suggestion
    for dataset_split_name, dm in {**dms, DatasetSplitName.all: dms}.items():
        if dataset_split_name not in dataset_split_names:
            continue
```
```diff
-            dm = dms[dataset_split_name] if dataset_split_name != DatasetSplitName.all else dms
```

Reply:
Absolutely, thanks for taking the time to simplify!

Reply:
I changed it a bit because `mypy` complained, and I also realized with should not be sending `dms` with `DatasetSplitName.all`, no `dm` is required, so I'm just sending `None`.

Reply:
Ho cool! Good job!